### ISSUE TITLE
Display names automatically modified

### ISSUE THREAD

**Issue by franz-benjamin** [homebridge/HAP-NodeJS#852]
When publishing an accessory, the original display name is extended by a four-letter hash, which happens in the `publish` method in `Accessory.ts`:
```javascript
// adding some identifying material to our displayName
    this.displayName = this.displayName + " " + crypto.createHash('sha512')
      .update(info.username, 'utf8')
      .digest('hex').slice(0, 4).toUpperCase();
    this.getService(Service.AccessoryInformation)!.updateCharacteristic(Characteristic.Name, this.displayName);
```

This seems to be the intended behaviour.  However, I cannot find any information about why the hash is added.  I checked the HAP Specification (non-commercial version) but I might have missed the corresponding information.

My questions to you:
1. Why is the hash automatically added to the display name?
2. Is there a formal requirement to automatically add the hash?
3. Is there a way to stop this behaviour?  That is, to publish exactly the display name I have chosen?  Any ideas are welcome.

Thank you very much for your appreciated replies!

**Comment by Supereg** [comment]
1. The hash is added to add some identifying material to the Accessory name. This is somewhat important as the Accessory Name is used as the Instance name for the mdns dns-sd (bonjour) service advertisement (as the instance name must be unique on the network).
This **was** relatively important for version < 0.8.x as the old bonjour-hap couldn't do proper name conflict resolution and would just fail. Especially in the case of homebridge, where typically every instance was just name "Homebridge".
Only with version 0.8.x and the new ciao library automatic name changing is more or less robust. The hash was still kept to reduce the amount of possible name changes necessary (as currently homebridge doesn't correctly persist names which got changed as of a detected conflict on the network).

2. There is no **formal** requirement, nope.
3. Currently there is no easy (intended) way of stopping that.

Would you mind sharing why this is a "problem" or annoyance in your case?

**Comment by franz-benjamin** [comment]
Dear @Supereg,

Thank you very much for your appreciated and complete answer.

> Would you mind sharing why this is a "problem" or annoyance in your case?

I use HAP-NodeJS as a library only, that is, independent of Homebridge.  Using HAP-NodeJS, I create several accessories without a bridge, and each of these accessories shows up separately in my Home.app on my iPhone.  As I want to have nice names in the app, I have to rename each single accessory individually, which takes time and feels unnecessary.

I understand well that the identifying material can and certainly is useful in many situations, and I understand that it makes sense to have it as the default behaviour.  Wouldn't it though make sense to have an option that suppresses the addition of the identifying material?  This would not break any existing code and would provide some better control about the accessories name in situations where needed.

**Comment by franz-benjamin** [comment]
Any reply?  Would be strongly appreciated! ;)

**Comment by Supereg** [comment]
Sorry. Don’t worry I didn’t forget you.
Just a bit busy right now.

**Comment by franz-benjamin** [comment]
Don't worry, I didn't want to push you.  I just felt showing some activity here to show that I am still interested in a reply and am appreciating it would be nice to see.  Take your time, I know what it means to be busy.

**Comment by Supereg** [comment]
Thanks, appreciate your patience.
I'm currently thinking about just doing an option in the publish method, but yet undecided what the default will be, maybe we can even set the default to not add the identifying material (and manually turn it on in homebridge for now?).
Anyways I think I will create a solution with the upcoming 0.9.0 release

**Comment by franz-benjamin** [comment]
That was really quick!  Again, both the fact that you reply as well as the fact that you consider this proposal is highly appreciated.
By the way: HAP-NodeJS is great!  Thanks for your efforts.

**Comment by Supereg** [comment]
@franz-benjamin We changed the default value for `addIdentifyingMaterial` back to true, meaning identifying material will be included by default as it is seemingly causing issues as described in #856. But at least there is a config option now 😅

**Comment by mocnik-science** [comment]
@Supereg: Thank you for informing me.  Having the option is fine and sufficient, I think.  I did not fully understand #856.  Is the issue only related to old versions, or is the option `addIdentifyingMaterial` "safe" to use?

**Comment by Supereg** [comment]
The option is save to use.
The problem only occurs when **directly** upgrading from a version using the old bonjour-hap library for dns-sd. The library doesn't cleanup the mdns records on shutdown (thus the old name is still present in mdns caches).
As the previous beta would use a new name, those "old" records would not be updated and pretty much the devices like iPhone or your HomeHub would get confused.

So in short, the old bonjour-hap library is misbehaving which causes issues in combination with that change.

**Comment by mocnik-science** [comment]
Thanks again for your very quick and great answer!  Very much appreciated.
